native/src/main/java/io/ballerina/persist/components/TypeDescriptor.java

```diff
@@ -0,0 +1,224 @@
+                SyntaxTreeConstants.SYNTAX_TREE_SEMICOLON
+        );
+    }
+    public static QualifiedNameReferenceNode getQualifiedNameReferenceNode(String modulePrefix, String identifier) {
+        return NodeFactory.createQualifiedNameReferenceNode(
+                AbstractNodeFactory.createIdentifierToken(modulePrefix),
+                SyntaxTreeConstants.SYNTAX_TREE_COLON,
+                AbstractNodeFactory.createIdentifierToken(identifier + " ")
+        );
+    }
+
+    public static BuiltinSimpleNameReferenceNode getBuiltinSimpleNameReferenceNode(String name) {
+        SyntaxKind kind;
+        switch (name) {
+            case "int":
+                kind = SyntaxKind.INT_TYPE_DESC;
+                break;
+            case "float":
+                kind = SyntaxKind.DECIMAL_TYPE_DESC;
+                break;
+            case "boolean":
+                kind = SyntaxKind.BOOLEAN_TYPE_DESC;
+                break;
+            case "byte":
+                kind = SyntaxKind.BYTE_TYPE_DESC;
+                break;
+            case "var":
+                kind = SyntaxKind.VAR_TYPE_DESC;
+                break;
+            case "anydata":
+                kind = SyntaxKind.ANYDATA_TYPE_DESC;
+                break;
+
+            default:
+                kind = SyntaxKind.STRING_TYPE_DESC;
+        }
+        return NodeFactory.createBuiltinSimpleNameReferenceNode(
+                kind,
+                AbstractNodeFactory.createIdentifierToken(name + " ")
+        );
+    }
+
+    public static ArrayTypeDescriptorNode getArrayTypeDescriptorNode(String type) {
+        ArrayDimensionNode arrayDimensionNode = NodeFactory.createArrayDimensionNode(
+                SyntaxTreeConstants.SYNTAX_TREE_OPEN_BRACKET,
+                null,
+                SyntaxTreeConstants.SYNTAX_TREE_CLOSE_BRACKET
+        );
+        NodeList<ArrayDimensionNode> dimensionList = NodeFactory.createNodeList(arrayDimensionNode);
+        return NodeFactory.createArrayTypeDescriptorNode(
+                getBuiltinSimpleNameReferenceNode(type),
+                dimensionList
+        );
+    }
+
+    public static RecordTypeDescriptorNode getRecordTypeDescriptorNode() {
+        return NodeFactory.createRecordTypeDescriptorNode(
+                SyntaxTreeConstants.SYNTAX_TREE_KEYWORD_RECORD,
+                SyntaxTreeConstants.SYNTAX_TREE_OPEN_BRACE,
+                NodeFactory.createEmptyNodeList(),
+                null,
+                SyntaxTreeConstants.SYNTAX_TREE_CLOSE_BRACE
+        );
+    }
+
+    public static RecordTypeDescriptorNode getClosedRecordTypeDescriptorNode() {
+        return NodeFactory.createRecordTypeDescriptorNode(
+                SyntaxTreeConstants.SYNTAX_TREE_KEYWORD_RECORD,
+                SyntaxTreeConstants.SYNTAX_TREE_OPEN_CLOSED_BRACE,
+                NodeFactory.createEmptyNodeList(),
+                null,
+                SyntaxTreeConstants.SYNTAX_TREE_CLOSE_CLOSED_BRACE
+        );
+    }
```

Comment:
```suggestion
```

Reply:
updated in [6d8b4a4](https://github.com/ballerina-platform/persist-tools/pull/15/commits/6d8b4a41b9a0ea1002bb906bb2008380cfa7cdb1)